src/devices/Gpio/Drivers/Sunxi/SunxiDriver.cs

```diff
@@ -0,0 +1,476 @@
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the driver's logical numbering scheme.</param>
+        /// <param name="eventTypes">The event types to wait for.</param>
+        /// <param name="cancellationToken">The cancellation token of when the operation should stop waiting for an event.</param>
+        /// <returns>A task representing the operation of getting the structure that contains the result of the waiting operation</returns>
+        protected override ValueTask<WaitForEventResult> WaitForEventAsync(int pinNumber, PinEventTypes eventTypes, CancellationToken cancellationToken)
+        {
+            _interruptDriver.OpenPin(pinNumber);
+            _pinModes[pinNumber].InUseByInterruptDriver = true;
+
+            return _interruptDriver.WaitForEventAsync(pinNumber, eventTypes, cancellationToken);
+        }
+
+        /// <summary>
+        /// Checks if a pin supports a specific mode.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the driver's logical numbering scheme.</param>
+        /// <param name="mode">The mode to check.</param>
+        /// <returns>The status if the pin supports the mode.</returns>
+        protected override bool IsPinModeSupported(int pinNumber, PinMode mode)
+        {
+            switch (mode)
+            {
+                case PinMode.Input:
+                case PinMode.InputPullDown:
+                case PinMode.InputPullUp:
+                case PinMode.Output:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        /// <summary>
+        /// Gets the mode of a pin.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the driver's logical numbering scheme.</param>
+        /// <returns>The mode of the pin.</returns>
+        protected override PinMode GetPinMode(int pinNumber)
+        {
+            if (!_pinModes.ContainsKey(pinNumber))
+            {
+                throw new InvalidOperationException("Can not get a pin mode of a pin that is not open.");
+            }
+
+            return _pinModes[pinNumber].CurrentPinMode;
+        }
+
+        /// <inheritdoc/>
+        protected override void Dispose(bool disposing)
+        {
+            if (_gpioPointer0 != default)
+            {
+                Interop.munmap(_gpioPointer0, 0);
+                _gpioPointer0 = default;
+            }
+
+            if (_gpioPointer1 != default)
+            {
+                Interop.munmap(_gpioPointer1, 0);
+                _gpioPointer1 = default;
+            }
+
+            if (_interruptDriver != default)
+            {
+                _interruptDriver.Dispose();
+                _interruptDriver = default;
+            }
+        }
+
+        private void InitializeInterruptDriver()
+        {
+            try
+            {
+                _interruptDriver = new LibGpiodDriver(0);
```

Comment:
I wish we had some better way of doing this.. might be worth filing issue.. we should instead have something like `GetBestDriverForBoard()` or something along the lines

Reply:
Actually `LinuxDriver.Create()` can solve this problem. But before solving this problem, I think we should solve `error CS1540` first. I can't access `LinuxDriver` methods.